server/src/controllers/Events/resolver.ts

```diff
@@ -402,12 +402,15 @@ ${unsubscribe}
     return event;
   }
 
-  // TODO: This will need a real GraphQL return type (AFAIK you have to return
-  // an object type)
-  @Mutation(() => Boolean)
-  async deleteEvent(@Arg('id', () => Int) id: number): Promise<boolean> {
-    await prisma.events.delete({ where: { id } });
-    return true;
+  @Mutation(() => Event)
+  async deleteEvent(@Arg('id', () => Int) id: number): Promise<Event> {
+    const result = await prisma.$transaction([
+      prisma.event_sponsors.deleteMany({ where: { event_id: id } }),
+      prisma.rsvps.deleteMany({ where: { event_id: id } }),
+      prisma.user_event_roles.deleteMany({ where: { event_id: id } }),
+      prisma.events.delete({ where: { id } }),
+    ]);
```

Comment:
How about doing this via the db schema?  We can attach the `@relation(...onDelete: Cascade)` attribute to the event_sponsors, rsvps and user_event_roles models and let prisma handle it.

I think it should either all be handled in resolver logic or all handled at the DB level.  We could go either way, but data integrity seems more the database's responsibility than the server's.

Reply:
Yep, that will make things easier in resolver. I've pushed changes with that.